File: Volume 1 Fundamental Algorithms/Euclid's algorithm/_.java
class algo {
    private static int euclidAlgoGCD(int m, int n) {
        return n == 0 ? m : euclidAlgoGCD(n, m % n);
    }

    public static void main(String[] args) {
        int n = 24, m = 36;
        /*
        IO
         */
        if (m < n) {
            int tmp = m;
            m = n;
            n = tmp;
        }
        System.out.println(euclidAlgoGCD(m, n));
    }
}

File: Volume 1 Fundamental Algorithms/Euclid's algorithm/_.py
# m, n = map(int, input().split())
# if m < n:
# 	m, n = n, m


def euclidAlgoGCD(m: int, n: int):
	return m if n == 0 else euclidAlgoGCD(n, m % n)

# print(eulerAlgoGCD(m, n))
